delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/plugin/PluginStatus.scala

```diff
@@ -0,0 +1,33 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.plugin
+
+import ch.epfl.bluebrain.nexus.delta.sdk.plugin.PluginStatus.Status
+
+/**
+  * Representation of a [[Plugin]] status.
+  *
+  * @param name    name of the [[Plugin]]
+  * @param version version of the [[Plugin]]
+  * @param status  status of the [[Plugin]]
+  */
+final case class PluginStatus(name: String, version: String, status: Status) {}
+
+object PluginStatus {
+
+  /**
+    * Plugin status.
+    */
+  sealed trait Status extends Product with Serializable
+  object Status {
```

Comment:
Starting and Stopping could be also interesting statuses too.

And as we can stop them maybe splitting down in 2, one for errors and one when stopping has been asked

Reply:
And if the plugin failed, it would be nice to have a reason

Reply:
starting and stopping would be interesting, but I think it complicates things... do you foresee a situation where a plugin has a long initialization?

Reply:
If communication with an external component takes time because this one is down for whatever reason.
Because most of the stuff should already be running.

But adding them could be added later when we will have more time or need to refine lifecycle for monitoring purposes for example.